contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -602,7 +589,16 @@ static VALUE rb_trilogy_connect(VALUE self, VALUE encoding, VALUE charset, VALUE
         connopt.tls_max_version = NUM2INT(val);
     }
 
-    int rc = try_connect(ctx, &handshake, &connopt);
+    VALUE io = rb_io_get_io(raw_socket);
+
+    rb_io_t *fptr;
+    GetOpenFile(io, fptr);
+    rb_io_check_readable(fptr);
+    rb_io_check_writable(fptr);
+
+    int fd = rb_io_descriptor(io);
```

Comment:
`rb_io_descriptor` was added relatively recently. Can't remember if it's 3.0 or 3.1.

Reply:
3.1 it seems. Fallback code if you want: https://github.com/Shopify/pitchfork/blob/e8e3626603c1ca5adbc53e4e5de9fc1581ea1483/ext/pitchfork_http/epollexclusive.h#L25-L33

Reply:
Thanks!